bcda/models/models_test.go

```diff
@@ -555,50 +555,149 @@ func (s *ModelsTestSuite) TestGetMaxBeneCount() {
+	assert.Equal(25, len(beneficiaries))
 
 	// Large ACO has 100 benes
 	acoUUID = uuid.Parse(constants.LARGEACOUUID)
 	err = s.db.Find(&largeACO, "UUID = ?", acoUUID).Error
 	assert.Nil(err)
-	beneficiaryIDs, err = largeACO.GetBeneficiaryIDs()
+	beneficiaries, err = largeACO.GetBeneficiaries(true)
 	assert.Nil(err)
-	assert.NotNil(beneficiaryIDs)
-	assert.Equal(100, len(beneficiaryIDs))
+	assert.NotNil(beneficiaries)
+	assert.Equal(100, len(beneficiaries))
 
 }
 
+func (s *ModelsTestSuite) TestGetBeneficiaries_Unsuppressed() {
+	acoCMSID := "T0000"
+	aco := ACO{UUID: uuid.NewRandom(), CMSID: &acoCMSID}
+	err := s.db.Save(&aco).Error
+	if err != nil {
+		s.FailNow("Failed to save ACO", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&aco)
+
+	cclfFile := CCLFFile{CCLFNum: 8, ACOCMSID: acoCMSID}
+	err = s.db.Save(&cclfFile).Error
+	if err != nil {
+		s.FailNow("Failed to save CCLF file", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&cclfFile)
+
+	// Beneficiary 1: preference indicator = N, effective date < now
+	bene1 := CCLFBeneficiary{FileID: cclfFile.ID, HICN: "bene1hicn"}
+	err = s.db.Save(&bene1).Error
+	if err != nil {
+		s.FailNow("Failed to save beneficiary", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene1)
+
+	bene1Suppression := Suppression{HICN: "bene1hicn", PrefIndicator: "N", EffectiveDt: time.Now().Add(-1 * time.Hour)}
+	err = s.db.Save(&bene1Suppression).Error
+	if err != nil {
+		s.FailNow("Failed to save suppression", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene1Suppression)
+
+	// Beneficiary 2: preference indicator = Y
+	bene2 := CCLFBeneficiary{FileID: cclfFile.ID, HICN: "bene2hicn"}
+	err = s.db.Save(&bene2).Error
+	if err != nil {
+		s.FailNow("Failed to save beneficiary", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene2)
+
+	bene2Suppression := Suppression{HICN: "bene2hicn", PrefIndicator: "Y"}
+	err = s.db.Save(&bene2Suppression).Error
+	if err != nil {
+		s.FailNow("Failed to save suppression", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene2Suppression)
+
+	// Beneficiary 3: no suppression record
+	bene3 := CCLFBeneficiary{FileID: cclfFile.ID, HICN: "bene3hicn"}
+	err = s.db.Save(&bene3).Error
+	if err != nil {
+		s.FailNow("Failed to save beneficiary", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene3)
+
+	// Beneficiary 4: preference indicator = N, effective date > now
+	bene4 := CCLFBeneficiary{FileID: cclfFile.ID, HICN: "bene4hicn"}
+	err = s.db.Save(&bene4).Error
+	if err != nil {
+		s.FailNow("Failed to save beneficiary", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene4)
+
+	bene4Suppression := Suppression{HICN: "bene4hicn", PrefIndicator: "N", EffectiveDt: time.Now().Add(1 * time.Hour)}
+	err = s.db.Save(&bene4Suppression).Error
+	if err != nil {
+		s.FailNow("Failed to save suppression", err.Error())
+	}
+	defer s.db.Unscoped().Delete(&bene4Suppression)
+
+	// Beneficiary 5: two suppression records, latest: preference indicator = N, effective date < now
```

Comment:
I think we might need more test cases here . (multiple suppression records).  I think it will be very common that a beneficiary will have multiple suppression records.

Reply:
I added one for `Y`, `N`, `Y`. Any others you can think of?

Reply:
Thanks!  I think that will cover us.